vscode.proposed.d.ts

```diff
@@ -0,0 +1,539 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
Is this our code or is it something that will eventually be a part of VSCode?

Reply:
> This requires checking in a vscode.proposed.d.ts file that includes the typings for Notebook APIs

Yeah it will eventually be a part of VS Code but until it's in stable, it has to be leveraged this way.